### PR TITLE
mds: enqueue ~mdsdir at the time of enqueing root

### PR DESCRIPTION
**Pull request by dparmar18**
This would avoid the need to run individual scrubs for
~mdsdir and root, i.e. run both the scrubs under the
same header, this also helps to avoid edge case where
in case ~mdsdir is huge and it's taking time to scrub it,
the scrub status would report something like this until
root inodes kick in:

```
{
    "status": "scrub active (757 inodes in the stack)",
    "scrubs": {}
}
```

Fixes: https://tracker.ceph.com/issues/59350

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [x] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by dparmar18** [comment]
If in case there are thousands of inodes to scrub in mdsdir, it is going to take some time for the scrub at root to end up in scrubbing_map i.e. scrub status would show something like this initially
```
{
    "status": "scrub active (757 inodes in the stack)",
    "scrubs": {}
}
```
until scrub for root kicks in, im currently trying to figure out a way such that the scrub status is not empty even at time of scrubbing mdsdir at root.

**Comment by dparmar18** [comment]
working as expected, run all green(except one failure where job failed due to some warning but tests went fine) - http://pulpito.front.sepia.ceph.com/dparmar-2023-05-23_00:32:04-fs:functional-wip-59350-distro-default-smithi/

**Review by vshankar — COMMENTED** [review]
Otherwise LGTM.

File: src/mds/ScrubStack.cc [hunk 1]
```diff
@@ -119,7 +119,20 @@ int ScrubStack::enqueue(CInode *in, ScrubHeaderRef& header, bool top)
 	     << ", conflicting tag " << header->get_tag() << dendl;
     return -CEPHFS_EEXIST;
   }
-
+  if(header->get_scrub_mdsdir()) {
```
[the remark this answers is not in the document]
Comment (dparmar18) [reply src/mds/ScrubStack.cc]: done

**Review by vshankar — COMMENTED** [review]
LGTM

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#18-July-2023

**Pull request merged**